taxi_ml/taxi_ml_benchmark.py

```diff
@@ -0,0 +1,306 @@
+    data_types_2014 = OrderedDict(
+        [
+            (" tolls_amount", "float64"),
+            (" surcharge", "float64"),
+            (" store_and_fwd_flag", "object"),
+            (" tip_amount", "float64"),
+            ("tolls_amount", "float64"),
+        ]
+    )
+
+    data_types_2015 = OrderedDict([("extra", "float64"), ("tolls_amount", "float64")])
+
+    data_types_2016 = OrderedDict([("tip_amount", "float64"), ("tolls_amount", "float64")])
+
+    # Dictionary of required columns and their datatypes
+    # Convert to list just to be clear that we only need column names, but keep types just in case
+    keep_cols = list(
+        {
+            "pickup_datetime": "datetime64[s]",
+            "dropoff_datetime": "datetime64[s]",
+            "passenger_count": "int32",
+            "trip_distance": "float32",
+            "pickup_longitude": "float32",
+            "pickup_latitude": "float32",
+            "rate_code": "int32",
+            "dropoff_longitude": "float32",
+            "dropoff_latitude": "float32",
+            "fare_amount": "float32",
+        }
+    )
+
+    dfs = []
+    for name, dtypes, date_cols in (
+        ("2014", data_types_2014, [" pickup_datetime", " dropoff_datetime"]),
+        ("2015", data_types_2015, ["tpep_pickup_datetime", "tpep_dropoff_datetime"]),
+        ("2016", data_types_2016, ["tpep_pickup_datetime", "tpep_dropoff_datetime"]),
+    ):
+        dfs.extend(
+            [
+                clean(
+                    read_csv(
+                        dirpath / filename,
+                        parse_dates=date_cols,
+                        col2dtype=dtypes,
+                        is_omniscidb_mode=is_omniscidb_mode,
+                    ),
+                    keep_cols,
+                )
+                for filename in islice((dirpath / name).iterdir(), 2 if debug else None)
+            ]
+        )
+
+    # concatenate multiple DataFrames into one bigger one
+    pd = get_pd()
+    df = pd.concat(dfs, ignore_index=True)
+
+    # To trigger execution
+    trigger_execution(df)
+
+    return df
+
+
+@measure_time
+def filter_df(df):
+    # apply a list of filter conditions to throw out records with missing or outlier values
+    df = df.query(
+        "(fare_amount > 1) & \
+        (fare_amount < 500) & \
+        (passenger_count > 0) & \
+        (passenger_count < 6) & \
+        (pickup_longitude > -75) & \
+        (pickup_longitude < -73) & \
+        (dropoff_longitude > -75) & \
+        (dropoff_longitude < -73) & \
+        (pickup_latitude > 40) & \
+        (pickup_latitude < 42) & \
+        (dropoff_latitude > 40) & \
+        (dropoff_latitude < 42) & \
+        (trip_distance > 0) & \
+        (trip_distance < 500) & \
+        ((trip_distance <= 50) | (fare_amount >= 50)) & \
+        ((trip_distance >= 10) | (fare_amount <= 300)) & \
+        (dropoff_datetime > pickup_datetime)"
+    )
+
+    return trigger_execution(df.reset_index(drop=True))
```

Comment:
isn't `trigger_execution` returns `None`? if so, we can't return its result from the `filter_df` function as further steps expect to receive a dataframe, not None.
```suggestion
    df = df.reset_index(drop=True)
    trigger_execution(df)
    return df
```